clients/drcachesim/tools/syscall_mix.cpp

```diff
@@ -117,14 +117,17 @@ syscall_mix_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
 #ifdef X64
         assert(static_cast<uintptr_t>(syscall_num) == memref.marker.marker_value);
 #endif
-        ++shard->syscall_counts[syscall_num];
+        ++shard->stats.syscall_counts[syscall_num];
     } else if (memref.marker.type == TRACE_TYPE_MARKER &&
                memref.marker.marker_type == TRACE_MARKER_TYPE_SYSCALL_TRACE_START) {
         int syscall_num = static_cast<int>(memref.marker.marker_value);
 #ifdef X64
         assert(static_cast<uintptr_t>(syscall_num) == memref.marker.marker_value);
 #endif
-        ++shard->syscall_trace_counts[syscall_num];
+        ++shard->stats.syscall_trace_counts[syscall_num];
+    } else if (memref.marker.type == TRACE_TYPE_MARKER &&
```

Comment:
Perhaps it's time to turn this into a switch guarded by a common TRACE_TYPE_MARKER condition.

Reply:
Done.